src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v1/mongo/EcommerceTransactionDataProvider.kt

```diff
@@ -186,4 +195,80 @@ class EcommerceTransactionDataProvider(
                 )
         }
     }
+
+    fun retrieveTransactionDetails(
```

Comment:
instead of implementing this method, can't we reuse `findResult()` to retrieve transaction information?

Reply:
Totally agree with the above comment :D 
Moreover I would avoid use a tuple here: code is not self explicative returning a `Mono<NTuple4<String, String, String, PaymentMethod>>` and from a caller point of view to understand what the second parameter is I would need to jump into method implementation detail.
Maybe a data class with explicative attribute names would fit better here

Reply:
Thanks for the suggestions. Both are resolved with [670ac72](https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/156/commits/670ac726123f1efd9d67ce01696b2e0212740f20). 

The findResult function has been reused and its result values have been used to call the getOrder, so there is no more need for the custom data class.